presto-main/src/test/java/com/facebook/presto/operator/aggregation/noisyaggregation/TestNoisySumGaussianDoubleAggregation.java

```diff
@@ -0,0 +1,419 @@
+                "Test noisy_sum_gaussian(double, noiseScale, lower, upper) with noiseScale > 0 which means some noise",
+                new Page(
+                        createDoublesBlock(values),
+                        createRLEBlock(DEFAULT_TEST_STANDARD_DEVIATION, numRows),
+                        createRLEBlock(lower, numRows),
+                        createRLEBlock(upper, numRows)),
+                expected);
+    }
+
+    @Test
+    public void testNoisySumGaussianDoubleClippingSomeNoiseScaleWithinSomeStd()
+    {
+        JavaAggregationFunctionImplementation noisySumGaussian = getFunction(DOUBLE, DOUBLE, DOUBLE, DOUBLE);
+
+        BiFunction<Object, Object, Boolean> withinSomeStdDoubleAssertion = (actual, expected) -> {
+            double actualValue = new Double(actual.toString());
+            double expectedValue = new Double(expected.toString());
+            return expectedValue - 5 * DEFAULT_TEST_STANDARD_DEVIATION <= actualValue && actualValue <= expectedValue + 5 * DEFAULT_TEST_STANDARD_DEVIATION;
+        };
+
+        int numRows = 10;
+        List<Double> values = createTestValues(numRows, true, 1.0, false);
+        double lower = 2.0;
+        double upper = 8.0;
+        double expected = clippedSum(values, lower, upper);
+        assertAggregation(
+                noisySumGaussian,
+                withinSomeStdDoubleAssertion,
+                "Test noisy_sum_gaussian(double, noiseScale, lower, upper) within some std from mean",
+                new Page(
+                        createDoublesBlock(values),
+                        createRLEBlock(DEFAULT_TEST_STANDARD_DEVIATION, numRows),
+                        createRLEBlock(lower, numRows),
+                        createRLEBlock(upper, numRows)),
+                expected);
+    }
+
+    // Test BIGINT with clipping and randomSeed
+    @Test
+    public void testNoisySumGaussianDoubleClippingFixedSignRandomSeed()
+    {
+        // Test DOUBLE with clipping where noise added changed the sign and need to be fixed
+        JavaAggregationFunctionImplementation noisySumGaussian = getFunction(DOUBLE, DOUBLE, DOUBLE, DOUBLE, BIGINT);
+
+        int numRows = 10;
+        List<Double> values = createTestValues(numRows, false, 1.0, false);
+        double lower = 2.0;
+        double upper = 8.0;
+        double expected = 0;
+        assertAggregation(
+                noisySumGaussian,
+                equalDoubleAssertion,
+                "Test noisy_sum_gaussian(double, noiseScale, lower, upper, randomSeed) that needs to fix sign",
+                new Page(
+                        createDoublesBlock(values),
+                        createRLEBlock(10000.0, numRows),
+                        createRLEBlock(lower, numRows),
+                        createRLEBlock(upper, numRows),
+                        createRLEBlock(23, numRows)), // this seed = 23 expected to create a negative random noise but lower > 0 => expect 0
```

Comment:
A question for one of the Presto maintainers: is it generally considered safe to write unit tests that rely on a seeded PRNG returning a specific result? There are a few such tests in here.

Reply:
I think that's a good way to generate deterministic tests when checking for expected results. 
Would be good to also exercise the non-deterministic path, even if we don't check for exact result matches